test/e2e-helm/e2e_helm_olm_test.go

```diff
@@ -15,25 +15,105 @@
+		const makefilePackagemanifests = `
+packagemanifests: kustomize
+	$(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests --version $(VERSION)
+`
 		BeforeEach(func() {
-			By("Turning off interactive prompts for all generation tasks.")
+			By("turning off interactive prompts for all generation tasks.")
 			replace := "operator-sdk generate kustomize manifests"
 			testutils.ReplaceInFile(filepath.Join(tc.Dir, "Makefile"), replace, replace+" --interactive=false")
 		})
 
 		AfterEach(func() {
+			By("destroying the deployed package manifests-formatted operator")
+			cleanupPkgManCmd := exec.Command(tc.BinaryName, "cleanup", "packagemanifests",
+				"--operator-version", operatorVersion,
+				"--timeout", "4m")
+			_, _ = tc.Run(cleanupPkgManCmd)
+
+			By("uninstalling CRD's")
+			_ = tc.Make("uninstall")
 		})
 
-		It("Should allow generate the OLM bundle and run it", func() {
+		It("should generate and run a valid OLM bundle and packagemanifests", func() {
+			By("building the bundle")
+			err := tc.Make("bundle")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating the bundle")
+			bundleValidateCmd := exec.Command(tc.BinaryName, "bundle", "validate", "./bundle")
+			_, err = tc.Run(bundleValidateCmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("building the operator bundle image")
+			// Use the existing image tag but with a "-bundle" suffix.
+			imageSplit := strings.SplitN(tc.ImageName, ":", 2)
+			bundleImage := path.Join("quay.io", imageSplit[0]+"-bundle")
+			if len(imageSplit) == 2 {
+				bundleImage += ":" + imageSplit[1]
+			}
+			err = tc.Make("bundle-build", "BUNDLE_IMG="+bundleImage)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("loading the project image into Kind cluster")
+			err = tc.LoadImageToKindClusterWithName(bundleImage)
+			Expect(err).Should(Succeed())
+
+			// We will add the target packagemanifests in the Makefile to  be able to run it in tests
+			// as the users would do it as well which indeed would be harder to executed directly via
+			// a cmd generated since it requires the kustomize which can be located in the bin of the project
+			// or in any place in the machine.
+			By("Adding to Makefile target to generate the packagemanifests")
+			makefileBytes, err := ioutil.ReadFile(filepath.Join(tc.Dir, "Makefile"))
+			Expect(err).Should(Succeed())
+
+			makefileBytes = append([]byte(makefilePackagemanifests), makefileBytes...)
+			err = ioutil.WriteFile(filepath.Join(tc.Dir, "Makefile"), makefileBytes, 0644)
+			Expect(err).Should(Succeed())
+
+			By("generating the operator package manifests")
+			err = tc.Make("packagemanifests")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating clusterserviceversion with the manager image")
+			testutils.ReplaceInFile(
+				filepath.Join(tc.Dir, "packagemanifests", operatorVersion,
+					fmt.Sprintf("e2e-%s.clusterserviceversion.yaml", tc.TestSuffix)),
+				"controller:latest", tc.ImageName)
```

Comment:
I understand the reasoning behind doing this, but wrapping the same command flow that you'd call in the CLI is pointless. You may as well copy the [Go e2e code](https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite.go#L182-L192) directly.

Reply:
Hi @estroz, 

The implementation suggested will not work for Helm/Ansible and actually IMO we need to change it in GO as well.

Following a few reasons:
a) See that you are calling [tc.KustomizeBuild(filepath.Join("config", "manifests"))](https://github.com/operator-framework/operator-sdk/blob/0c7d59bf34747d305a8ebbd54113ac6330b42239/test/internal/utils.go#L57) it will NOT work for helm/ansible. This projects did not install kustomize with go get since go is not a pre-requirement for them. 

b) See my comment on top of that: 

```
// We will add the target packagemanifests in the Makefile to  be able to run it in tests
// as the users would do it as well which indeed would be harder to executed directly via
// a cmd generated since it requires the kustomize which can be located in the bin of the project
// or in any place in the machine.
```

b) IMO the changes made here are easier to understand as well.

Reply:
Helm and Ansible projects will need to install the `kustomize` binary regardless of whether Go is present or not, and this is reflected in their Makefiles. The exact same commands should work for Go, Ansible, Helm:
```sh
$ operator-sdk init
$ operator-sdk create api
$ operator-sdk generate kustomize manifests
$ ${KUSTOMIZE} build config/manifests | operator-sdk generate packagemanifests --version ${VERSION}
```

Reply:
HI @estroz, 

Yes, I agree that the same commands should work for all. However, note that:

- The Go implementation  [Go e2e code](https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite.go#L182-L192) is NOT doing `${KUSTOMIZE} build config/manifests` it is doing `kustomize build config/manifests`
- We should use the makefile targets scaffolded by default instead of calling the commands in order to keep its understand easily and we are able to indeed test them. (same doc steps if possible) It can prevent issues. See that the GO tests are just passing because they are not using the makefile targets Otherwise, it would fail with an error when the `docker-build` would be called. 
- Call `kustomize build config/manifests` as it is done in [Go e2e code](https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite.go#L182-L192) just work for GO because it was installed by `go get dep` which is not possible for Helm/Ansible. 

Sadly. in this specific case I am reluctant to address your suggestion because:
- It will not work for Ansible/Helm since we can not use GO  to install kustomize
- Indeed, IMO we need to change the GO impl because it is harder to understand and keep maintained, not using  the makefile targets, do not work locally when the kustomize is installed in another way)
- IMO we suggest to users customize the Makefile when they need (e.g in this scenario). So, we could do the test following our suggestion which is how it was done. 

I hope that clarifies my motivation here.

Reply:
Sounds like we should be installing kustomize into some element of $PATH for Helm and Ansible operators, or locally to `bin/kustomize` for Go operators. Behavior should be consistent.

Reply:
I understand that the behaviour is consistent when we use the Makefile target with `${KUSTOMIZE}` since it will get the kustomize installed. My idea as follow up would cleanup the Go tests as well and ensure that all types are following the same go-tests structure and doing the same common tests.

Reply:
Hi @estroz, 

See that: 
- I updated it to be == we suggest to do it in the docs
- Added the docs links to make clear that
- Add a note in the docs for users know that for ansible and helm the target is == but without the manifests.

Reply:
Can we just update $PATH in these tests for now (in `BeforeSuite`) and address your concerns in a follow-up? I agree we should make sure command usage is fairly consistent across all plugins, which requires work outside of the scope of this PR potentially.

Reply:
> Can we just update $PATH in these tests to include ./bin for now (in BeforeSuite)`
No, because the kustomize can be or not in the ./bin and the test should run locally with diff setups. 

Also, see that in order to make consistent we have the GO as well: #3499. Why we would make it work as GO if it is the Go test is not accurate and consistent with the steps described in the docs as it is not working locally?

Reply:
We can run `make kustomize` as a pre-step. At that point `kustomize` is either in `bin/kustomize` OR in the $PATH.

I think @estroz is suggesting that if we add `./bin` to $PATH, then we guarantee that after running `make kustomize`, `kustomize` will be in $PATH (either existing in the user's local setup, or in `./bin`, which has been added to $PATH).

Reply:
Hi @joelanford @estroz, 
 
1 - Why we would call setup the bin PATH for just this test? Again, the kustomzie can or NOT be installed in the bin. The test should work locally.
2 - Why we would call `kustomize build config/manifests` instead to do the test as we suggested for users to do in the docs?

Reply:
Ok so I think we agree that we _can_ do it without appending to the Makefile and running make targets.

The question now is, should we append some targets to the Makefile and run a make target, or should we just make the calls directly.

1. If we use the Makefile approach, we have to add some targets to the Makefile, but then we get the KUSTOMIZE path logic "for free".
1. If we use the Makefile approach, we are testing the [approach that we suggest to users in our docs](https://master.sdk.operatorframework.io/docs/olm-integration/generation/#package-manifests-format).
1. If we use the direct approach, we have to add `<projectDir>/bin` to the PATH.
1. If we use the direct approach, we can call the commands directly, which has less indirection and would make the test code more obvious about what's actually being tested.

Reply:
If we're going with the Makefile approach in this PR, I'd like a function to exist that updates the Makefile with some code snippet by appending. The current test code is quite cluttered by this Makefile update.

Reply:
IMO appending in the makefile also:
- Makes it easier to read and understand the code and reduces the complexity
- Also, see that by following the suggestion we are not applying the same options suggested in the docs (e.g  see `$(PKG_MAN_OPTS)`)

So, usually, I address all suggestions made but this one I still reluctant because I was not convinced that it will bring advantages and then, I hope that we can reach a consensus that we can keep it as it is. :-)

Reply:
Closing it since shows solved.